### ISSUE TITLE
Valdie is throwing error in the run time which crashes the app

### ISSUE THREAD

**Issue by PatrickNiyogitare28** [Rwanda-Coding-Academy/Valdie#6]
## Bug report
The library works fine when the provided parameter is a valid type, but it throws the error if the value is not of a valid type. This crushes the app.

<h5>index.js</h5>

```
const vald = require('valdie');
const email = "test@gmail.com";

const isValid = vald.isEmail(email);
console.log(isValid)
```

If we change the ``email`` to be an invalid email the error will be thrown in the app hence crashing the app

```
const vald = require('valdie');
const email = "testgmail.com.com";

const isValid = vald.isEmail(email);
console.log(isValid)
```

<h2>Error log</h2>

```
C:\Users\patri\Documents\workspaces\OPEN\test-valdie\node_modules\valdie\dist\lib\utils\Error.js:4
    throw new Error(`${value} is not a valid ${key}`);
    ^

Error: patrickgmail.com is not a valid email
    at Object.ErrorMessage [as default] (C:\Users\patri\Documents\workspaces\OPEN\test-valdie\node_modules\valdie\dist\lib\utils\Error.jsPS C:\Users\patri\Documents\workspaces\OPEN\test-valdie> node .\index.js
C:\Users\patri\Documents\workspaces\OPEN\test-valdie\node_modules\valdie\dist\lib\utils\Error.js:4
    throw new Error(`${value} is not a valid ${key}`);
    ^
PS C:\Users\patri\Documents\workspaces\OPEN\test-valdie> node .\index.js
C:\Users\patri\Documents\workspaces\OPEN\test-valdie\node_modules\valdie\dist\lib\utils\Error.js:4
    throw new Error(`${value} is not a valid ${key}`);
    ^

Error: patrickgmail.com is not a valid email
    at Object.ErrorMessage [as default] (C:\Users\patri\Documents\workspaces\OPEN\test-valdie\node_modules\valdie\dist\lib\utils\Error.js:4:11)
    at Object.isEmail (C:\Users\patri\Documents\workspaces\OPEN\test-valdie\node_modules\valdie\dist\lib\isEmail.js:11:24)
    at Object.<anonymous> (C:\Users\patri\Documents\workspaces\OPEN\test-valdie\index.js:5:19)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
PS C:\Users\patri\Documents\workspaces\OPEN\test-valdie>
```

**Comment by makuzaverite** [comment]
What will be a good solution? 

I was thinking of  an idea what if we return a error message and let developers have handle,  it however they want to

like 

```
cosnt {error,message} = vald.isEmail(email)

if(error){
 // error message goes here
}

if( message){
// handle for collectness here
}

```

**Comment by PatrickNiyogitare28** [comment]
That would be great!

**Comment by claranceliberi** [comment]
> What will be a good solution?
> 
> I was thinking of an idea what if we return a error message and let developers have handle, it however they want to
> 
> like
> 
> ```
> cosnt {error,message} = vald.isEmail(email)
> 
> if(error){
>  // error message goes here
> }
> 
> if( message){
> // handle for correctness here
> }
> ```

This is awesome, but what if we use world `valid` instead of `error`, I think it would be more meaningful and it could be good name according to this package name
```javascript
const {valid,message} = vald.isEmail(email)

 if(!valid){
  // error message goes here
  }
 if( message){
 // handle for correctness here
 }
```

**Comment by makuzaverite** [comment]
That would be great what do you think @PatrickNiyogitare28

**Comment by claranceliberi** [comment]
guys how are you far with this issue? 
about last proposal or error returning could this work  ?
```typescript
export default function ErrorMessage(value: unknown, key: unknown): Response {
  return {
    valid: false,
    message: `${value} is not a valid ${key}`
  } as Response
}
``` 

but I was also asking myself what are going to return when the value is valid? 
if we keep it in the old way the `valid` will be falsy since it will be not returned which will read to falsy response for all 
and also I was proposing this new way 
```typescript 
//response
export default function Res(): Response {
  return {
    valid: true
  } as Response
}
```

but also got doubt on it, because it not a good experience for the user compared to returning `true` when value is valid

**Comment by PatrickNiyogitare28** [comment]
Yeah you got a point, let's find a proper way.

**Comment by veritem** [comment]
i've setup a new method of error handling and should be fixed by `v0.1.0`
